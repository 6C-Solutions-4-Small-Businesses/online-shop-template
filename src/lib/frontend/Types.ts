import type {AddressPresentation} from '$lib/frontend/presentations/AddressPresentation'

export type AddressUpdatedEvent = { detail: { address: AddressPresentation } }
