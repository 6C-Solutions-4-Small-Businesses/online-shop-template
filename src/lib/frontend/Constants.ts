export const API_BASE_ENDPOINT = '/api/v1'
export const BASE_HEADERS = {
    'Content-Type': 'application/json',
    'X-App-Source': 'SvelteKit'
}
