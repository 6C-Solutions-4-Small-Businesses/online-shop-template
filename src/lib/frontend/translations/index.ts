import { dev } from '$app/environment';
import type { Config } from 'sveltekit-i18n';
import i18n from 'sveltekit-i18n';
import lang from './lang.json';

export const config: Config = {
	log: {
		level: dev ? 'warn' : 'error'
	},
	translations: {
		en: { lang },
		fr: { lang }
	},
	loaders: [
		{
			locale: 'en',
			key: 'home',
			loader: async () => (await import('./en/home.json')).default
		},
		{
			locale: 'fr',
			key: 'home',
			loader: async () => (await import('./fr/home.json')).default
		}
	]
};

export const {
	t,
	loading,
	locales,
	locale,
	translations,
	loadTranslations,
	addTranslations,
	setLocale,
	setRoute
} = new i18n(config);

loading.subscribe(($loading) => $loading && console.log('Loading translations...'));
