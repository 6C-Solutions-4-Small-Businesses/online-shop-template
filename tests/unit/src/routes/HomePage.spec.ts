import {render} from '@testing-library/svelte'
import {beforeAll, describe, expect} from 'vitest'
import HomePage from '$routes/+page.svelte'
import {IntersectionObserver} from '$mocks/src/lib/frontend/IntersectionObserverStub'

describe('Home page', () => {
    beforeAll(() => {

        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: vi.fn(),
        })
        window.IntersectionObserver = IntersectionObserver
        global.IntersectionObserver = IntersectionObserver
        window.ResizeObserver =
            window.ResizeObserver ||
            vi.fn().mockImplementation(() => ({
                disconnect: vi.fn(),
                observe: vi.fn(),
                unobserve: vi.fn(),
            }))
    })

    it('should have match snapshot', () => {
        const {container} = render(HomePage, {
            data: {
                collections: [
                    {
                        id: 'collection-id-1',
                        name: 'spotlight',
                        description: 'a-collection',
                        type: 'Spotlight',
                        offers: [
                            {
                                id: 'product-id-1',
                                name: 'product-1',
                                price: 100,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Centiliter',
                            },
                            {
                                id: 'product-id-2',
                                name: 'product-2',
                                price: 800,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Milliliter',
                            },
                            {
                                id: 'product-id-3',
                                name: 'product-3',
                                price: 1200,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Gramme',
                            }
                        ]
                    },
                    {
                        id: 'collection-id-2',
                        name: 'En sp√©ciale',
                        description: 'a-collection',
                        type: 'Sale',
                        offers: [
                            {
                                id: 'product-id-4',
                                name: 'product-4',
                                price: 100,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Unit',
                            },
                            {
                                id: 'product-id-5',
                                name: 'product-5',
                                price: 800,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Pound',
                            },
                            {
                                id: 'product-id-6',
                                name: 'product-6',
                                price: 1200,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Portion',
                            }
                        ]
                    },
                    {
                        id: 'collection-id-3',
                        name: 'Nous vous recommandons',
                        description: 'a-collection',
                        type: 'Common',
                        offers: [
                            {
                                id: 'product-id-7',
                                name: 'product-7',
                                price: 100,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Kilogram',
                            },
                            {
                                id: 'product-id-8',
                                name: 'product-8',
                                price: 800,
                                image: 'an-image-url',
                                description: 'a-product',
                                quantity: 10,
                                isSoldByQuantities: false,
                                unit: 'Kilogram',
                            }
                        ]
                    }
                ]
            }
        })

        expect(container).toMatchSnapshot()
    })
})
