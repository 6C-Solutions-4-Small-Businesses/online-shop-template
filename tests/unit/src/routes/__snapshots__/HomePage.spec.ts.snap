// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > should have match snapshot 1`] = `
<body>
  <div>
    <div
      class="s-y_bCXRrkrYfP"
    >
      <img
        alt="Demo MERCHANT, Business Slogan"
        class="object-cover w-full max-w-full h-screen brightness-75 blur-xs relative -z-1 s-y_bCXRrkrYfP"
        src="/src/lib/frontend/assets/images/location-interior.webp"
      />
       
      <div
        class="absolute top-44 w-full flex justify-center s-y_bCXRrkrYfP"
      >
        <div
          class="embla w-10/12 md:w-2/3 xl:w-1/3 s-y_bCXRrkrYfP"
        >
          <div
            class="embla__container s-y_bCXRrkrYfP"
            style="transform: translate3d(0px,0px,0px);"
          >
            <div
              class="embla__slide w-2/3 md:w-2/3 xl:w-1/3 s-y_bCXRrkrYfP"
            >
              <div
                class="w-full relative flex flex-col bg-white"
                role="none"
              >
                <img
                  alt="product-1"
                  class="object-contain h-[15rem] sm:h-[30rem] md:h-[30rem] lg:h-[30rem] xl:h-[30rem] py-5"
                  src="an-image-url"
                />
                 
                <div
                  class="h-16 flex items-center justify-center gap-2 text-3xl rounded-none bg-white"
                >
                  <span>
                    product-1
                  </span>
                   
                  <span
                    class="font-bold text-red-500"
                  >
                    $
                    1
                  </span>
                </div>
                 
                <div
                  class="hidden"
                >
                  <div
                    class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                    role="region"
                  >
                     
                    <div
                      class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                    >
                      <svg
                        height="1.2em"
                        viewBox="0 0 24 24"
                        width="1.2em"
                      >
                        <path
                          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                          fill="currentColor"
                        />
                      </svg>
                      <!--&lt;Cart-plus&gt;-->
                    </div>
                  </div>
                  <!--&lt;ShoppingCartActions&gt;-->
                </div>
              </div>
              <!--&lt;SpotlightedProductCard&gt;-->
               
            </div>
            <div
              class="embla__slide w-2/3 md:w-2/3 xl:w-1/3 s-y_bCXRrkrYfP"
            >
              <div
                class="w-full relative flex flex-col bg-white"
                role="none"
              >
                <img
                  alt="product-2"
                  class="object-contain h-[15rem] sm:h-[30rem] md:h-[30rem] lg:h-[30rem] xl:h-[30rem] py-5"
                  src="an-image-url"
                />
                 
                <div
                  class="h-16 flex items-center justify-center gap-2 text-3xl rounded-none bg-white"
                >
                  <span>
                    product-2
                  </span>
                   
                  <span
                    class="font-bold text-red-500"
                  >
                    $
                    8
                  </span>
                </div>
                 
                <div
                  class="hidden"
                >
                  <div
                    class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                    role="region"
                  >
                     
                    <div
                      class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                    >
                      <svg
                        height="1.2em"
                        viewBox="0 0 24 24"
                        width="1.2em"
                      >
                        <path
                          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                          fill="currentColor"
                        />
                      </svg>
                      <!--&lt;Cart-plus&gt;-->
                    </div>
                  </div>
                  <!--&lt;ShoppingCartActions&gt;-->
                </div>
              </div>
              <!--&lt;SpotlightedProductCard&gt;-->
               
            </div>
            <div
              class="embla__slide w-2/3 md:w-2/3 xl:w-1/3 s-y_bCXRrkrYfP"
            >
              <div
                class="w-full relative flex flex-col bg-white"
                role="none"
              >
                <img
                  alt="product-3"
                  class="object-contain h-[15rem] sm:h-[30rem] md:h-[30rem] lg:h-[30rem] xl:h-[30rem] py-5"
                  src="an-image-url"
                />
                 
                <div
                  class="h-16 flex items-center justify-center gap-2 text-3xl rounded-none bg-white"
                >
                  <span>
                    product-3
                  </span>
                   
                  <span
                    class="font-bold text-red-500"
                  >
                    $
                    12
                  </span>
                </div>
                 
                <div
                  class="hidden"
                >
                  <div
                    class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                    role="region"
                  >
                     
                    <div
                      class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                    >
                      <svg
                        height="1.2em"
                        viewBox="0 0 24 24"
                        width="1.2em"
                      >
                        <path
                          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                          fill="currentColor"
                        />
                      </svg>
                      <!--&lt;Cart-plus&gt;-->
                    </div>
                  </div>
                  <!--&lt;ShoppingCartActions&gt;-->
                </div>
              </div>
              <!--&lt;SpotlightedProductCard&gt;-->
               
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="w-full s-y_bCXRrkrYfP"
      >
        <div
          class="w-full pt-8 pb-8 bg-green-50 s-Lrs6oclqAYHN"
        >
          <div
            class="w-full flex justify-center text-3xl collection-name not-italic font-light mb-5 s-Lrs6oclqAYHN"
          >
            En sp√©ciale
          </div>
           
          <div
            class="w-full p-5 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 3xl:grid-cols-6 s-Lrs6oclqAYHN"
          >
            <div
              class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl s-0QAyY96I1Nkt"
              role="none"
            >
              <div
                class="absolute flex w-12 h-12 top-0 text-white items-center invisible s-0QAyY96I1Nkt"
              >
                <svg
                  class="s-0QAyY96I1Nkt"
                  height="100"
                  viewBox="0 0 490 490"
                  width="100"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <polygon
                    class="s-0QAyY96I1Nkt"
                    points="0,0 490,0 0,490"
                    style="fill: #ff0000;"
                  />
                </svg>
                 
                <div
                  class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold s-0QAyY96I1Nkt"
                >
                  -
                  null
                  %
                </div>
              </div>
               
              <img
                alt="product-4"
                class="product-image object-contain absolute h-full w-full top-0 p-4 s-0QAyY96I1Nkt"
                src="an-image-url"
              />
               
              <div
                class="w-full absolute flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                  role="region"
                >
                   
                  <div
                    class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                  >
                    <svg
                      height="1.2em"
                      viewBox="0 0 24 24"
                      width="1.2em"
                    >
                      <path
                        d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                        fill="currentColor"
                      />
                    </svg>
                    <!--&lt;Cart-plus&gt;-->
                  </div>
                </div>
                <!--&lt;ShoppingCartActions&gt;-->
              </div>
               
              <div
                class="absolute bottom-0 right-0 w-full flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg s-0QAyY96I1Nkt"
                  role="region"
                >
                   
                  <div
                    class="w-full flex flex-col s-0QAyY96I1Nkt"
                  >
                     
                    <div
                      class="w-full items-center justify-center flex text-xl font-semibold s-0QAyY96I1Nkt"
                    >
                      <span
                        class="block s-0QAyY96I1Nkt"
                      >
                        $
                        1
                        <sup
                          class="s-0QAyY96I1Nkt"
                        >
                          00
                        </sup>
                      </span>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!--&lt;ProductCard&gt;-->
            <div
              class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl s-0QAyY96I1Nkt"
              role="none"
            >
              <div
                class="absolute flex w-12 h-12 top-0 text-white items-center invisible s-0QAyY96I1Nkt"
              >
                <svg
                  class="s-0QAyY96I1Nkt"
                  height="100"
                  viewBox="0 0 490 490"
                  width="100"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <polygon
                    class="s-0QAyY96I1Nkt"
                    points="0,0 490,0 0,490"
                    style="fill: #ff0000;"
                  />
                </svg>
                 
                <div
                  class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold s-0QAyY96I1Nkt"
                >
                  -
                  null
                  %
                </div>
              </div>
               
              <img
                alt="product-5"
                class="product-image object-contain absolute h-full w-full top-0 p-4 s-0QAyY96I1Nkt"
                src="an-image-url"
              />
               
              <div
                class="w-full absolute flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                  role="region"
                >
                   
                  <div
                    class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                  >
                    <svg
                      height="1.2em"
                      viewBox="0 0 24 24"
                      width="1.2em"
                    >
                      <path
                        d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                        fill="currentColor"
                      />
                    </svg>
                    <!--&lt;Cart-plus&gt;-->
                  </div>
                </div>
                <!--&lt;ShoppingCartActions&gt;-->
              </div>
               
              <div
                class="absolute bottom-0 right-0 w-full flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg s-0QAyY96I1Nkt"
                  role="region"
                >
                   
                  <div
                    class="w-full flex flex-col s-0QAyY96I1Nkt"
                  >
                     
                    <div
                      class="w-full items-center justify-center flex text-xl font-semibold s-0QAyY96I1Nkt"
                    >
                      <span
                        class="block s-0QAyY96I1Nkt"
                      >
                        $
                        8
                        <sup
                          class="s-0QAyY96I1Nkt"
                        >
                          00
                        </sup>
                      </span>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!--&lt;ProductCard&gt;-->
            <div
              class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl s-0QAyY96I1Nkt"
              role="none"
            >
              <div
                class="absolute flex w-12 h-12 top-0 text-white items-center invisible s-0QAyY96I1Nkt"
              >
                <svg
                  class="s-0QAyY96I1Nkt"
                  height="100"
                  viewBox="0 0 490 490"
                  width="100"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <polygon
                    class="s-0QAyY96I1Nkt"
                    points="0,0 490,0 0,490"
                    style="fill: #ff0000;"
                  />
                </svg>
                 
                <div
                  class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold s-0QAyY96I1Nkt"
                >
                  -
                  null
                  %
                </div>
              </div>
               
              <img
                alt="product-6"
                class="product-image object-contain absolute h-full w-full top-0 p-4 s-0QAyY96I1Nkt"
                src="an-image-url"
              />
               
              <div
                class="w-full absolute flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                  role="region"
                >
                   
                  <div
                    class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                  >
                    <svg
                      height="1.2em"
                      viewBox="0 0 24 24"
                      width="1.2em"
                    >
                      <path
                        d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                        fill="currentColor"
                      />
                    </svg>
                    <!--&lt;Cart-plus&gt;-->
                  </div>
                </div>
                <!--&lt;ShoppingCartActions&gt;-->
              </div>
               
              <div
                class="absolute bottom-0 right-0 w-full flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg s-0QAyY96I1Nkt"
                  role="region"
                >
                   
                  <div
                    class="w-full flex flex-col s-0QAyY96I1Nkt"
                  >
                     
                    <div
                      class="w-full items-center justify-center flex text-xl font-semibold s-0QAyY96I1Nkt"
                    >
                      <span
                        class="block s-0QAyY96I1Nkt"
                      >
                        $
                        12
                        <sup
                          class="s-0QAyY96I1Nkt"
                        >
                          00
                        </sup>
                      </span>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!--&lt;ProductCard&gt;-->
          </div>
           
          <div
            class="w-full flex justify-center items-center text-xl collection-name not-italic font-light h-10 mt-5 s-Lrs6oclqAYHN"
          >
            <button
              class="hover:text-green-500 s-Lrs6oclqAYHN"
            >
              Afficher Plus
            </button>
          </div>
        </div>
        <!--&lt;Collection&gt;-->
        
         
        <div
          class="news-letter-box w-full flex flex-col items-center xl:flex-row xl:justify-center bg-green-100 s-y_bCXRrkrYfP"
        >
          <div
            class="flex flex-col w-10/12 xl:w-7/12 py-12 xl:py-20 s-y_bCXRrkrYfP"
          >
            <div
              class="flex flex-col s-y_bCXRrkrYfP"
            >
              <div
                class="text-3xl text-green-900 my-2 s-y_bCXRrkrYfP"
              >
                Rabais Exclusifs
              </div>
               
              <p
                class="font-thin text-justify s-y_bCXRrkrYfP"
              >
                En tant qu'abonn√© privil√©gi√©, vous b√©n√©ficierez d'un acc√®s exclusif √† des offres sp√©ciales et √†
                        des
                        promotions, vous assurant de r√©aliser plus d'√©conomies √† chaque fois que vous faites vos achats
              </p>
            </div>
             
            <div
              class="flex flex-col s-y_bCXRrkrYfP"
            >
              <div
                class="text-3xl text-green-900 my-2 s-y_bCXRrkrYfP"
              >
                Premier Arriv√©, Premier Servi
              </div>
               
              <p
                class="font-thin text-justify s-y_bCXRrkrYfP"
              >
                Abonnez-vous d√®s maintenant et soyez le premier √† savoir quand nous r√©approvisionnons en
                        produits frais,
                        produits laitiers et autres denr√©es p√©rissables, vous garantissant ainsi de toujours obtenir les
                        choix les
                        plus frais de nos rayons.
              </p>
            </div>
          </div>
           
          <div
            class="w-10/12 xl:w-3/12 flex flex-col justify-center items-center s-y_bCXRrkrYfP"
          >
            <div
              class="w-80 h-80 flex flex-col items-center bg-green-300 rounded-full justify-center s-y_bCXRrkrYfP"
            >
              <div
                class="w-4/5 mt-5 mb-8 text-2xl text-center text-green-900 font-thin s-y_bCXRrkrYfP"
              >
                Abonnez-vous √† notre
                        Infolettre
              </div>
               
              <input
                class="w-4/5 h-10 border border-gray-500  rounded-sm pl-2 focus:outline-none  focus:border-2 s-y_bCXRrkrYfP"
                placeholder="Entrer votre courriel"
                type="email"
              />
               
              <button
                class="flex justify-around items-center uppercase rounded-sm font-bold w-36 shadow-sm border border-orange-500 text-orange-500 bg-orange-50 h-12 hover:bg-white mt-10 bg-white hover:bg-orange-50 border text-black"
                data-testid=""
                id=""
              >
                S'abonner
              </button>
              <!--&lt;Button&gt;-->
            </div>
          </div>
        </div>
         
        <div
          class="w-full pt-8 pb-8 bg-green-50 s-Lrs6oclqAYHN"
        >
          <div
            class="w-full flex justify-center text-3xl collection-name not-italic font-light mb-5 s-Lrs6oclqAYHN"
          >
            Nous vous recommandons
          </div>
           
          <div
            class="w-full p-5 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 3xl:grid-cols-6 s-Lrs6oclqAYHN"
          >
            <div
              class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl s-0QAyY96I1Nkt"
              role="none"
            >
              <div
                class="absolute flex w-12 h-12 top-0 text-white items-center invisible s-0QAyY96I1Nkt"
              >
                <svg
                  class="s-0QAyY96I1Nkt"
                  height="100"
                  viewBox="0 0 490 490"
                  width="100"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <polygon
                    class="s-0QAyY96I1Nkt"
                    points="0,0 490,0 0,490"
                    style="fill: #ff0000;"
                  />
                </svg>
                 
                <div
                  class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold s-0QAyY96I1Nkt"
                >
                  -
                  null
                  %
                </div>
              </div>
               
              <img
                alt="product-7"
                class="product-image object-contain absolute h-full w-full top-0 p-4 s-0QAyY96I1Nkt"
                src="an-image-url"
              />
               
              <div
                class="w-full absolute flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                  role="region"
                >
                   
                  <div
                    class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                  >
                    <svg
                      height="1.2em"
                      viewBox="0 0 24 24"
                      width="1.2em"
                    >
                      <path
                        d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                        fill="currentColor"
                      />
                    </svg>
                    <!--&lt;Cart-plus&gt;-->
                  </div>
                </div>
                <!--&lt;ShoppingCartActions&gt;-->
              </div>
               
              <div
                class="absolute bottom-0 right-0 w-full flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg s-0QAyY96I1Nkt"
                  role="region"
                >
                   
                  <div
                    class="w-full flex flex-col s-0QAyY96I1Nkt"
                  >
                     
                    <div
                      class="w-full items-center justify-center flex text-xl font-semibold s-0QAyY96I1Nkt"
                    >
                      <span
                        class="block s-0QAyY96I1Nkt"
                      >
                        $
                        1
                        <sup
                          class="s-0QAyY96I1Nkt"
                        >
                          00
                        </sup>
                      </span>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!--&lt;ProductCard&gt;-->
            <div
              class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl s-0QAyY96I1Nkt"
              role="none"
            >
              <div
                class="absolute flex w-12 h-12 top-0 text-white items-center invisible s-0QAyY96I1Nkt"
              >
                <svg
                  class="s-0QAyY96I1Nkt"
                  height="100"
                  viewBox="0 0 490 490"
                  width="100"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <polygon
                    class="s-0QAyY96I1Nkt"
                    points="0,0 490,0 0,490"
                    style="fill: #ff0000;"
                  />
                </svg>
                 
                <div
                  class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold s-0QAyY96I1Nkt"
                >
                  -
                  null
                  %
                </div>
              </div>
               
              <img
                alt="product-8"
                class="product-image object-contain absolute h-full w-full top-0 p-4 s-0QAyY96I1Nkt"
                src="an-image-url"
              />
               
              <div
                class="w-full absolute flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
                  role="region"
                >
                   
                  <div
                    class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
                  >
                    <svg
                      height="1.2em"
                      viewBox="0 0 24 24"
                      width="1.2em"
                    >
                      <path
                        d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                        fill="currentColor"
                      />
                    </svg>
                    <!--&lt;Cart-plus&gt;-->
                  </div>
                </div>
                <!--&lt;ShoppingCartActions&gt;-->
              </div>
               
              <div
                class="absolute bottom-0 right-0 w-full flex justify-end s-0QAyY96I1Nkt"
              >
                <div
                  class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg s-0QAyY96I1Nkt"
                  role="region"
                >
                   
                  <div
                    class="w-full flex flex-col s-0QAyY96I1Nkt"
                  >
                     
                    <div
                      class="w-full items-center justify-center flex text-xl font-semibold s-0QAyY96I1Nkt"
                    >
                      <span
                        class="block s-0QAyY96I1Nkt"
                      >
                        $
                        8
                        <sup
                          class="s-0QAyY96I1Nkt"
                        >
                          00
                        </sup>
                      </span>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!--&lt;ProductCard&gt;-->
          </div>
           
          <div
            class="w-full flex justify-center items-center text-xl collection-name not-italic font-light h-10 mt-5 s-Lrs6oclqAYHN"
          >
            <button
              class="hover:text-green-500 s-Lrs6oclqAYHN"
            >
              Afficher Plus
            </button>
          </div>
        </div>
        <!--&lt;Collection&gt;-->
        
      </div>
    </div>
    <!--&lt;+page&gt;-->
  </div>
</body>
`;
